irepela/homework_2/sum_numbers.py

```diff
@@ -0,0 +1,31 @@
+def sum_numbers(text: str) -> int:
+    """
+        Extracts numbers from the text and calculates their sum
+
+        Args:
+            text (str): text to extract numbers from
+
+        Returns:
+            int: Sum of the numbers extracted from the text
+    """
+    text_list = text.split()
+    numbers_total = 0
+    for item in text_list:
+        if item.isnumeric():
+            numbers_total += int(item)
+    return numbers_total
+
+
+def sum_numbers_comprehension(text: str) -> int:
+    """
+        Extracts numbers from the text and calculates their sum
+
+        Args:
+            text (str): text to extract numbers from
+
+        Returns:
+            int: Sum of the numbers extracted from the text
+    """
+    text_list = text.split()
+    numbers_total = sum([int(item) for item in text_list if item.isnumeric()])
```

Comment:
no need to create intermediate list by list comprehension, 
you can use generator expression, which does not create intermediate list, but provides items 1 by 1 on demand

```
    numbers_total = sum((int(item) for item in text_list if item.isnumeric()))

# if generator expression is the only input parameter, parenthesis can be avoided

    numbers_total = sum(int(item) for item in text_list if item.isnumeric())

```

Reply:
at the end of the day solution can be converted to 1 liner:

```
return   sum(int(item) for item in text.split() if item.isnumeric())
```